File: app/src/main/java/com/plants/InformationsFragment.java
package com.plants;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import com.plants.databinding.FragmentInformationsBinding;
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.List;
import java.util.stream.Collectors;

public class InformationsFragment extends Fragment {
    private InformationsAdapter adapter;
    private FragmentInformationsBinding binding;
    private List<Integer> checkedChipIds;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        adapter = new InformationsAdapter();
        binding = FragmentInformationsBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        if (binding != null) {
            checkedChipIds = binding.cgFragmentInformations.getCheckedChipIds();
            populateRecyclerView();

            binding.cgFragmentInformations.setOnCheckedStateChangeListener(
                    (group, checkedIds) -> {
                        checkedChipIds = checkedIds;
                        populateRecyclerView();
                    });

            binding.rvFragmentDiscussions.setAdapter(adapter);
            binding.rvFragmentDiscussions.setLayoutManager(
                    new LinearLayoutManager(getContext()));
        }
    }

    private void populateRecyclerView() {
        adapter.removeAll();

        if (getContext() != null && !checkedChipIds.isEmpty()) {
            String fileName;
            int chipId = checkedChipIds.get(0);

            if (chipId == R.id.cFragmentInformationsDesert) {
                fileName = "desert.json";
            } else if (chipId == R.id.cFragmentInformationsTemperate) {
                fileName = "temperate.json";
            } else if (chipId == R.id.cFragmentInformationsWetland) {
                fileName = "wetland.json";
            } else {
                fileName = "tropical.json";
            }

            try {
                BufferedReader reader = new BufferedReader(
                        new InputStreamReader(getContext().getAssets().open(fileName)));
                String jsonContent = reader.lines().collect(Collectors.joining("\n"));
                reader.close();

                Information[] informations = Utils.getGson()
                        .fromJson(jsonContent, Information[].class);

                for (Information information : informations) {
                    adapter.add(information);
                }
            } catch (IOException e) {
                e.printStackTrace();
            }
        }

        if (binding != null) {
            binding.tvFragmentDiscussions.setVisibility(
                    adapter.getItemCount() > 0 ? View.GONE : View.VISIBLE);
        }
    }
}

File: app/src/main/java/com/plants/InformationsFragment.kt
package com.plants

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import com.plants.databinding.FragmentInformationsBinding
import okio.buffer
import okio.source

class InformationsFragment : Fragment() {
    private lateinit var adapter: InformationsAdapter
    private var binding: FragmentInformationsBinding? = null
    private var checkedChipIds: List<Int> = mutableListOf()

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        adapter = InformationsAdapter()
        binding = FragmentInformationsBinding.inflate(inflater, container, false)
        return binding!!.root
    }

    override fun onDestroyView() {
        super.onDestroyView()
        binding = null
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        binding?.apply {
            checkedChipIds = cgFragmentInformations.checkedChipIds
            populateRecyclerView()

            cgFragmentInformations.setOnCheckedStateChangeListener { _, checkedIds ->
                checkedChipIds = checkedIds
                populateRecyclerView()
            }

            rvFragmentDiscussions.also {
                it.adapter = adapter
                it.layoutManager = LinearLayoutManager(context)
            }
        }
    }

    private fun populateRecyclerView() {
        adapter.removeAll()

        when (checkedChipIds[0]) {
            R.id.cFragmentInformationsDesert -> "desert.json"
            R.id.cFragmentInformationsTemperate -> "temperate.json"
            R.id.cFragmentInformationsWetland -> "wetland.json"
            else -> "tropical.json"
        }.also {
            context?.apply {
                getGson().fromJson(
                    assets.open(it).source().buffer().readUtf8(),
                    Array<Information>::class.java
                ).forEach { adapter.add(it) }
            }
        }

        binding?.tvFragmentDiscussions?.visibility = if (adapter.itemCount > 0) {
            View.GONE
        } else {
            View.VISIBLE
        }
    }
}